packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1127,8 +1127,32 @@ abstract class State<T extends StatefulWidget> with Diagnosticable {
   /// ```
   ///
   /// It is an error to call this method after the framework calls [dispose].
-  /// You can determine whether it is legal to call this method by checking
+  /// You can determine whether it is legal to call [setState] by checking
   /// whether the [mounted] property is true.
+  ///
+  /// # Performance considerations
+  ///
+  /// Calls to [setState] cause a rebuild of the current widget and,
+  /// potentially, all of its children. This isn't efficient and generally not
+  /// desiderable. A [setState] call should only be made on widgets with small
+  /// subtrees in order to rebuild the least amount of children.
+  ///
+  /// A better alternative to [setState] is represented by the combined usage
+  /// of a [Listenable] along with an [AnimatedBuilder], which is:
+  ///
+  ///   1. Performant. A [setState] call could entirely rebuild the subtree of
+  ///      a widget. A [Listenable] instead can be used along with
+  ///      [AnimatedBuilder] to only rebuild some pieces of the widget, leaving
+  ///      other parts and its children untouched.
+  ///   2. Predictable. There is no way to know how many widgets will be
```

Comment:
Isn't this basically the same thing as performance? You really only care about how many widgets rebuild because it may hurt performance.